feat: add GeoDNS documentation

deansheather opened the pull request.

deansheather commented:
Very rough. I'm not super happy with the Cloudflare section taking up the whole document but I'm unsure how to solve it

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+---
+title: GeoDNS
```
How about we call this "global load balancing," or something similar?

If you google GeoDNS, this is the top hit: https://en.wikipedia.org/wiki/GeoDNS

Do these docs only refer to Anycast routing, or can people use this if they are using split-horizon DNS?

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+---
+
+Out of the box, satellites deployments have a different access URL to the
+primary deployment. This may cause confusion for engineering teams located
```
This says "This may cause confusion" twice

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+should access Coder from.
+
+Coder supports a GeoDNS configuration where all satellites and the primary
+deployment share the same hostname, so all users access Coder using a consistent
```
needs a passage about users being routed to the nearest installation or satellite, ensuring low latency

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+
+## Configure GeoDNS on Coder
+
+To configure GeoDNS onn Coder:
```
```suggestion
To configure GeoDNS on Coder:
```

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+
+To configure GeoDNS onn Coder:
+
+1. In the primary helm values file set `coderd.alternateHostnames` to your
```
```suggestion
1. In the primary helm values file, set `coderd.alternateHostnames` to your
```

jawnsy commented on admin/satellites/geo-dns.md:
````diff
@@ -0,0 +1,120 @@
+           - "coder.example.com"
+       ```
+
+1. Deploy your primary and satellites with your new helm values.
````
```suggestion
1. Deploy your primary and satellites with your new Helm values.
```

jawnsy commented on admin/satellites/geo-dns.md:
````diff
@@ -0,0 +1,120 @@
+   ```
+
+1. In each satellite helm values file:
+    1. Set `coderd.satellite.accessURL` to your GeoDNS URL (this value is used
````
```suggestion
    1. Set `coderd.satellite.accessURL` to your primary access URL, which is the name of the global load balancer in front of Coder deployments (this value is used
```

jawnsy commented on admin/satellites/geo-dns.md:
```diff
@@ -0,0 +1,120 @@
+- A primary access URL (e.g. `https://primary.example.com`).
+- One or more satellite access URLs (e.g. `https://sydney.example.com`,
+  `https://london.example.com`).
+- A GeoDNS load balancer access URL (e.g. `https://coder.example.com`). The
```
GCP does something similar with Anycast, https://cloud.google.com/load-balancing

We should distinguish whether this feature is for people using Anycast (a CDN or global load balancing) versus split-horizon DNS (where the DNS responses vary according to the user's IP address/location)

khorne3 commented:
@deansheather I don't think the Cloudflare section is too big of a problem, but if you'd like, we can split it into its own guide? Then people could use if if they need the help in that section, otherwise they can ignore if they have a load balancer already

khorne3 commented:
@deansheather do you have any thoughts on @jawnsy 's suggestions?

- Calling this feature `global load balancing` or similar
- Distinguish whether this feature is for people using Anycast (a CDN or global load balancing) versus split-horizon DNS (where the DNS responses vary according to the user's IP address/location)

Other than that, I think all else is ok

deansheather commented:
@jawnsy GeoDNS is a correct term for it AFAIK and means the same thing as split-horizon DNS. What are your thoughts on disambiguating it in the first paragraph but keeping it called GeoDNS?

deansheather commented:
@khorne3

- I can rename it to "global load balancing" if you think it's a good idea, GeoDNS is a good name though IMO
- I added a paragraph that says it will work with both anycast and split-horizon

deansheather commented:
I've renamed it to "Proximity based configuration" but I don't think that's a good name yet. Also, I've replaced most of the references to "GeoDNS access URL" with "unified access URL" instead
